main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1664,6 +1664,21 @@ object Typer extends Phase[ResolvedAst.Program, TypedAst.Root] {
           elementType <- unifyTypAllowEmptyM(elementTypes,loc);
           resultType <- unifyTypM(tvar, mkArray(elementType), loc)
         ) yield resultType
+
+      case ResolvedAst.Pattern.ArrayTailSpread(elms, varSym, tvar, loc) =>
+        for (
+          elementTypes <- seqM(elms map visit);
+          elementType <- unifyTypAllowEmptyM(elementTypes,loc);
+          resultType <- unifyTypM(tvar, mkArray(elementType), loc)
+        ) yield resultType
+
+      case ResolvedAst.Pattern.ArrayHeadSpread(varSym, elms, tvar, loc) =>
+        for (
+          elementTypes <- seqM(elms map visit);
```

Comment:
You should use `varSym` somewhere. Otherwise you can write type incorrect programs. Try it!